dagmc_stats/DagmcStats.py

```diff
@@ -0,0 +1,257 @@
+            if aspect_ratio:
+                tri_data_row['aspect_ratio'] = np.prod(side_lengths) / (8 * p)
+        df_tri = df_tri.append(tri_data_row, ignore_index=True)
+        return
+
+    # Todo
+    def get_tris(self, meshset):
+        """Get triangles of a volume if geom_dim is 3
+        Get triangles of a surface if geom_dim is 2
+        Else get all the triangles
+
+        inputs
+        ------
+        meshset : set of entities that are used to populate data. Default value
+            is None and data of the whole geometry will be populated.
+
+        outputs
+        -------
+        tris : a list of triangle entities
+        """
+        dim = _my_moab_core.tag_get_data(dagmc_tags['geom_dim'], meshset)[0][0]
+        # get triangles of a volume or a surface
+        if dim == 3 or dim == 2:
+            tris = _my_moab_core.get_entities_by_type(meshset, types.MBTRI)
+        else:
+            # get all the triangles
+            tris = _my_moab_core.get_entities_by_type(root_set, types.MBTRI)
+        return tris
+
+    def get_tri_side_length(self, tri):
+        """
+        Get side lengths of triangle
+
+        inputs
+        ------
+        tri : triangle entity
+
+        outputs
+        -------
+        side_lengths : a dictionary that stores vert : the opposite side length of
+        the vert as key-value pair
+        """
+
+        side_lengths = {}
+        coord_list = []
+
+        verts = list(_my_moab_core.get_adjacencies(tri, 0))
+
+        for vert in verts:
+            coords = _my_moab_core.get_coords(vert)
+            coord_list.append(coords)
+
+        for side in range(3):
+            side_lengths.update({verts[side-1]:
+                                 np.linalg.norm(coord_list[side] -
+                                                coord_list[side-2])})
+            # Although it may not be intuitive, the indexing of these lists takes
+            # advantage of python's indexing syntax to rotate through
+            # the `verts` of the triangle while simultaneously referencing the side
+            # opposite each of the `verts` by the coordinates of the vertices that
+            # define that side:
+            #    side       side-1   index(side-1)     side-2   index(side-2)
+            #     0           -1           2             -2             1
+            #     1            0           0             -1             2
+            #     2            1           1              0             0
+        return side_lengths
+
+    # overload this somehow: either optional parameters, or different names
+
+    def query_triangle_area(self, tri):
+        """Query triangle areas given triangle entity handles
+
+        inputs
+        ------
+        tri : triangle entity handles
+
+        outputs
+        -------
+        triangle areas that correspond to the given triangle entity handles
+        """
+        return self._tri_data.query('tri_eh == tri')
```

Comment:
I don't know this syntax very well, but I think we need the @ symbol here
```suggestion
        return self._tri_data.query('tri_eh == @tri')
```

Reply:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.query.html